pkg/distro/arch/arch.go

```diff
@@ -296,6 +298,27 @@ func (d *arch) InstallPackages(ctx context.Context, c *cache.Cache, pkgs []files
 	return nil
 }
 
+var (
+	//go:embed Dockerfile.generate-hash.tmpl
+	dockerfileGenerateHashTmpl string
+
+	//go:embed Dockerfile.tmpl
+	dockerfileTmpl string
+)
+
 func (d *arch) GenerateDockerfile(ctx context.Context, dir string, args distro.DockerfileTemplateArgs, opts distro.DockerfileOpts) error {
-	return distro.ErrNotImplemented
+	if d.info.Name != NameArchLinux {
+		return fmt.Errorf("generating dockerfiles needs the distro driver to be set to %q, not %q", NameArchLinux, d.info.Name)
+	}
```

Comment:
Are you planning to add support for a fork of ArchLinux?
Otherwise this check is not necessary.

Reply:
I see, I've removed this change and also removed my changes regarding Name/NameArchLinux :)